requirements/testing.txt

```diff
@@ -85,6 +101,8 @@ parameterized==0.9.0
     # via -r requirements/testing.in
 pathable==0.4.3
     # via jsonschema-spec
+prophet==1.1.3
```

Comment:
A new version of prophet is already available :)

```suggestion
prophet==1.1.4
```

Reply:
If I think about it: `"prophet": ["prophet>=1.1.0, <2.0.0"],` would always install latest prophet 1.x, but CI would only test the pinned release. Maybe we should pin prophet in `setup.py`?

Reply:
@sebastianliebscher CI uses the pinned version in the frozen requirements.

Reply:
I mean, with future releases of prophet if a user enables the optional dependency with `pip install -e '.[prophet]'`, they would potentially install a version that is not tested against CI.

Reply:
The issue you describe is true for any package (or sub-package) which is not explicitly pinning.

Note we (Airbnb) typically install Superset as a service by:

1. Defining a `requirements/production.in` file which includes the lines

```
-e .[prophet,…]
ddtrace
…
```

2. Running `pip-compile-multi` to compile the dependencies—taking into account the `prophet`, `ddtrace`, etc. production dependencies.
3. Installing Superset via,

```
python3 -m pip install -r requirements/production.txt
python3 -m pip install -e .
```

which ensures we always install Superset in a deterministic manner which is super critical.

Why doesn’t Superset explicitly pin every package (and sub-package) in `setup.py`? Though packages should provide flexibility, one could argue this is valid for a service—given it typically isn’t used as a dependency—however we run into the issue of creating an infeasible dependency space when installation dependent requirements are added. By relaxing constraints (by way of removing explicit pins) we allow `pip-compile-multi` to solve this somewhat difficult problem on our behalf.

Reply:
A bit late, but I thank you very much for your detailed explanation! This helped a lot to understand how Superset manages dependencies.